pkg/bucketaccess/bucketaccess_controller.go

```diff
@@ -147,6 +155,7 @@ func (bal *BucketAccessListener) Add(ctx context.Context, inputBucketAccess *v1a
 	}
 
 	if bucket.Status.BucketReady != true || bucket.Status.BucketID == "" {
+		bal.recordEvent(inputBucketAccess, v1.EventTypeWarning, events.WaitingForBucket, "BucketAccess can't be granted to bucket not in Ready state and without a bucketID")
```

Comment:
From experience with Rook, we've found that it's quite a bit nicer to give specific context info when reporting things like this so users (or ourselves) don't have to inspect and cross reference as many resources when issues are coming up.

```suggestion
		bal.recordEvent(inputBucketAccess, v1.EventTypeWarning, events.WaitingForBucket, "BucketAccess can't be granted to Bucket %q not in Ready state (isReady? %b) or without a bucketID (ID empty? %b)", bucket.Name, bucket.Status.BucketReady, bucket.Status.BucketID=="")
```

Since bucket ID could be considered sensitive info by some users, I opted to leave the ID as true/false empty or not in my suggestion, so as not to leak the info.